release-notes/10.0/preview/preview7/aspnetcore.md

```diff
@@ -2,14 +2,250 @@
+
+For the majority of applications this should have no impact on behavior. However, if the application is using a custom `JsonConverter`, there is a chance that the converter doesn't handle [Utf8JsonReader.HasValueSequence](https://learn.microsoft.com/dotnet/api/system.text.json.utf8jsonreader.hasvaluesequence) correctly. This can result in missing data and errors like `ArgumentOutOfRangeException` when deserializing.
+
+The quick workaround (especially if you don't own the custom `JsonConverter` being used) is to set the `"Microsoft.AspNetCore.UseStreamBasedJsonParsing"` [AppContext](https://learn.microsoft.com/dotnet/api/system.appcontext) switch to `true`. This should be a temporary workaround and the `JsonConverter`(s) should be updated to support `HasValueSequence`.
+
+To fix `JsonConverter` implementations, there is the quick fix which allocates an array from the `ReadOnlySequence`:
+
+```csharp
+public override T? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+{
+    var span = reader.HasValueSequence ? reader.ValueSequence.ToArray() : reader.ValueSpan;
+    // previous code
+}
+```
+
+Or the more complicated (but performant) fix which would involve having a separate code path for the `ReadOnlySequence` handling:
+
+```csharp
+public override T? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+{
+    if (reader.HasValueSequence)
+    {
+        reader.ValueSequence;
+        // ReadOnlySequence optimized path
+    }
+    else
+    {
+        reader.ValueSpan;
+        // ReadOnlySpan optimized path
+    }
+}
+```
+
+## Enhanced validation for classes and records
+
+Users can now use validation attributes on both classes and records, with consistent code generation and validation behavior. This enhances flexibility when designing models using records in ASP.NET Core applications.
+
+**Community contribution: Thanks to [@marcominerva](https://github.com/marcominerva)**
+
+## Blazor component improvements
+
+### Resource preloader component renamed
+
+The Blazor component for rendering preloading links has been renamed from `LinkPreload` to `ResourcePreloader` based on API review feedback.
+
+### API review feedback implementation
+
+Blazor APIs have been updated to implement API review feedback, including:
+
+- Renamed JavaScript APIs: `Blazor.pause()` → `Blazor.pauseCircuit()` and `Blazor.resume()` → `Blazor.resumeCircuit()`
+- Renamed C# attribute: `SupplyParameterFromPersistentComponentStateAttribute` → `PersistentStateAttribute`
+- Updated related class names for consistency
+
+### Enhanced NotFound support
+
+Added support for `NotFound` in applications without Blazor's `Router`. Applications that implement their custom router can now use `NavigationManager.NotFound()`. The implementation also simplifies 404 handling with the default `Router` by making the `NotFound` fragment obsolete.
+
+### Diagnostic metrics and traces improvements
```

Comment:
Provide a table of the old and new metric and trace names so that users can upgrade.